options: don't report errors on help value for OPT_FLAG and OPT_COLOR

pavelxdd opened the pull request:
The same idea as in 3723e61 but for OPT_FLAG now.

Before the change:
-----
````
mpv --border=help
Invalid parameter for border flag: help
Valid values are:
    yes
    no
    (passing nothing)
Error parsing option border (option parameter could not be parsed)
Setting commandline option --border=help failed.

Exiting... (Fatal error)
````

````
mpv --border=sdjfkhlsdfh
Invalid parameter for border flag: sdjfkhlsdfh
Valid values are:
    yes
    no
    (passing nothing)
Error parsing option border (option parameter could not be parsed)
Setting commandline option --border=sdjfkhlsdfh failed.

Exiting... (Fatal error)
````

After the change:
-----
````
mpv --border=help
Valid values for border flag are:
    yes
    no
    (passing nothing)
````

````
mpv --border=sdjfkhlsdfh
Invalid parameter for border flag: sdjfkhlsdfh
Valid values are:
    yes
    no
    (passing nothing)
Error parsing option border (option parameter could not be parsed)
Setting commandline option --border=sdjfkhlsdfh failed.

Exiting... (Fatal error)
````

pavelxdd commented:
Added a commit implementing the same behavior for `OPT_COLOR`.

wm4 commented on options/m_option.c:
```diff
@@ -135,13 +135,18 @@ static int parse_flag(struct mp_log *log, const m_option_t *opt,
+        mp_info(log, "Valid values are:\n");
+    } else {
+        mp_info(log, "Valid values for %.*s flag are:\n", BSTR_P(name));
+    }
```
Better reorder if/else body, so it's `if (cond)` instead of `if (!cond)`

wm4 commented on options/m_option.c:
```diff
@@ -1792,10 +1796,11 @@ static int parse_color(struct mp_log *log, const m_option_t *opt,
-                "Or in the form 'r/g/b/a', where each component is a value in the\n"
-                "range 0.0-1.0. (Also allowed: 'gray', 'gray/a', 'r/g/b'.\n");
-    return M_OPT_INVALID;
+print_help:
```
Would probably be better to have only one label (at the position of the error label), and print the error only if `!is_help`?

wm4 commented:
LGTM, made some minor improvement suggestions.

pavelxdd commented:
Updated, thanks for suggestions.
Added a 3rd commit for `OPT_GEOMETRY`

Before the change:
-----
````
mpv --geometry=help
Option geometry: invalid geometry: 'help'
Valid format: [W[%][xH[%]]][{+-}X[%]{+-}Y[%]] | [X[%]:Y[%]]
Error parsing option geometry (option parameter could not be parsed)
Setting commandline option --geometry=help failed.

Exiting... (Fatal error)
````

After the change:
-----
````
mpv --geometry=help
Valid format: [W[%][xH[%]]][{+-}X[%]{+-}Y[%]] | [X[%]:Y[%]]
````

````
mpv --geometry=helpdsffds
Option geometry: invalid geometry: 'helpdsffds'
Valid format: [W[%][xH[%]]][{+-}X[%]{+-}Y[%]] | [X[%]:Y[%]]
Error parsing option geometry (option parameter could not be parsed)
Setting commandline option --geometry=helpdsffds failed.

Exiting... (Fatal error)
````

jeeb commented:
Passes *nix and Windows builds and generally looks good to me.
